src/JWT.Extensions.AspNetCore/JwtAuthenticationEvents.cs

```diff
@@ -0,0 +1,101 @@
+namespace JWT.Extensions.AspNetCore
+{
+
+    using System;
+    using Microsoft.AspNetCore.Authentication;
+    using Microsoft.Extensions.Logging;
+    using Microsoft.Net.Http.Headers;
+
+    public class JwtAuthenticationEvents
```

Comment:
`sealed`, please. Tiny but a perf gain

Reply:
So, we don't want this sealed because then we can inherit from it. That is the crux of the work. You inherit from the existing one. Then you override the method you are interested in to plug in your own logic.

Reply:
Ah, right! Makes sense.